### DISCUSSION TITLE
Zones need to be clearly defined as ether Camera Specific properties or Global Objects.

### DISCUSSION THREAD

**Discussion by bagobones** [blakeblackshear/frigate#13034] [Ideas]
I have a few problems with zones and I think the main issue is it is unclear if they are indented to be global objects or camera properties.

Argument for Camera:
- Per previous discussions while you can have more than one camera have the same zone name label counts are not designed for this.
- The new loiter directives would not work if it was global.

Argument for Global:
- Having one zone / place covered by many cameras is great! if ANY camera triggers an object in that zone from its point of view you can get an alert for that zone 
- It makes filtering really easy.. If I have more than one camera covering the driveway I can see all related clips with one filter.
- The current frigate home assistant integration generates zones as TOP LEVEL devices beside cameras as if they are global objects.

I think the frigate team needs to ether make zones global and support having multiple camera configs some how, or make them camera specific and put in validation failures when a user configures a zone with the same name.. it is currently VERY easy to do.

**Comment by NickM-27** [comment]
I think you are creating a false dichotomy here, there is no reason it needs to be one or the other. 

In general, global zones does not make sense because it would be extremely unlikely that the exact same coordinates for a zone would work in both camera cases. So, especially given that it all is done in the UI, zones should be configured at the camera level. However, I see no reason that you can't have the same zone name on multiple cameras and I have multiple cameras like this personally and it has never caused me any issue.

**Reply by bagobones** [reply]
>However, I see no reason that you can't have the same zone name on multiple cameras and I have multiple cameras like this personally and it has never caused me any issue.

you may want to look at the people counts and if loitering is working correctly on those zones if they are being shared.

Ok but then in the HA integration the presence, label counts etc treats them as global objects. And it has been already stated the code doesn't really intentionally handle the mixing of presence and counts in any intelligent way, it isn't intended.

**Reply by NickM-27** [reply]
> you may want to look at the people counts and if loitering is working correctly on those zones if they are being shared.

yeah, that is something that can be explicitly handled (and documented how it is handled) when that part of the code is refactored

**Reply by bagobones** [reply]
To be honest I would prefer if zones where global objects where a mix of both.

The zone has a presence and object count that is global and is designed to handle more than one camera with defined logic
- IE you could chose if count was was the [cameraname]/sum/avg/latest.  IE camera choses the one camera, sum is the total count, avg is the avg count between all cameras, latest is similar to the current which the latest update from any cam being the current.
- presence could have similar camera aware settings but maybe two separate settings one for trigger and one for reset. IE trigger could be cameraname/any/both. where camera is just one specific camera, any is if ether show , and both would require both cameras to trigger.. the reset value could be similar and or have a time out variable in seconds.

**Reply by bagobones** [reply]
In the current state things are ambiguous.. 

I think now that I have given separate names to my overlapping zones that loiter is working correctly.

And as for the counters and presence I don't use them.

I am using the over lapping zone names for filtering, which is why separating them for loiter is kind of annoying. Going back to this if zones where defined globally loiter could be defined once and then inherited by cameras with the zone.

another way to think of it is that zones at the global level would be meta objects and zones in the camera level would be camera specific. IE shape and triggering would be handled at the camera level then at the global level that would be rolled up differently depending on your global settings..

**Reply by NickM-27** [reply]
the loitering is handled on a per object per camera level, the zones on different cameras having the same name or not won't affect anything at all in the code

I still don't see a reason to make this config more complicated, I think it is working fine as is besides needing clarification / improvement on the HA integration side of things
